hi-ml/src/health_ml/utils/data_augmentations.py

```diff
@@ -171,17 +171,34 @@ def __init__(self, kernel_size: int, p: float = 0.5, min: float = 0.1, max: floa
         self.kernel_size = kernel_size
         self.p = p
 
-    def __call__(self, sample: torch.Tensor) -> torch.Tensor:
+    @staticmethod
+    def apply_gaussian_blur(sample: torch.Tensor, kernel_size: int, p: float, min: float, max: float) -> torch.Tensor:
         prob = np.random.random_sample()
-
-        if prob < self.p:
-            sigma = (self.max - self.min) * np.random.random_sample() + self.min
-            sample = np.array(sample.squeeze())  # type: ignore
-            sample = cv2.GaussianBlur(sample, (self.kernel_size, self.kernel_size), sigma)
-            sample = torch.Tensor(sample).unsqueeze(0)
-
+        if prob < p:
+            sigma = (max - min) * np.random.random_sample() + min
+            sample = sample.permute([0, 2, 3, 1]).squeeze().numpy()   # only 3 channels, color channel last
+            sample = cv2.GaussianBlur(sample, (kernel_size, kernel_size), sigma)
+            sample = torch.Tensor(sample).unsqueeze(0).permute(0, 3, 1, 2)
         return sample
 
+    def __call__(self, img: torch.Tensor) -> torch.Tensor:
+        original_shape = img.shape
+        if len(original_shape) == 3:
+            img = img.unsqueeze(0)  # add batch dimension if missing
+        # if the input is a bag of images, gaussian blur needs to run on each image separately
+        if img.shape[0] > 1:
+            for i in range(img.shape[0]):
+                img_tile = img[i]
+                img[i] = self.apply_gaussian_blur(sample=img_tile.unsqueeze(0), kernel_size=self.kernel_size, p=self.p,
+                                                  min=self.min, max=self.max)
+            return img
+        else:
+            img = self.apply_gaussian_blur(sample=img, kernel_size=self.kernel_size, p=self.p,
+                                           min=self.min, max=self.max)
+            if len(original_shape) == 3:
```

Comment:
might be out of scope but it would be nice to write a wrapper function or even transform to fix the reshaping, the same way we have `transform_dict_adaptor`

Reply:
exactly, I have noted this, we could address in a separate PR.